### PR TITLE
Run `apt update` after adding repo on jammy

### PR DESCRIPTION
**Pull request by jdblischak**
I was following the steps in `add_cranapt_jammy.sh` to setup r2u in a GitHub Actions build. Strangely it was installing way too many R packages, which I fixed by running `apt update`. This step is already included in `add_cranapt_focal.sh`, so I copied it to the jammy script.

### CODE REVIEW

**Comment by eddelbuettel** [comment]
Interesting.  But when you write 

> t was installing way too many R packages, which I fixed by running `apt update`. 

something does not mesh as `apt update` only updates the meta info.  It does not actually install anything.  Did you mean `apt upgrade -qq -y` or alike?

Of course, I would very much like to squash any bug you may have seen so a repro example for me to try in an (otherwise empty) Docker container would be great.   Otherwise I do have a few initial CI runs to where I injected `r2u` into the [r-ci](https://eddelbuettel.github.io/r-ci/) setup and plan to do something more here.  Contributions to fold `r2u` into more conventional 'Actions' are likely welcome too.

**Comment by jdblischak** [comment]
> something does not mesh as apt update only updates the meta info. It does not actually install anything. Did you mean apt upgrade -qq -y or alike?

Sorry for the imprecise wording. My point was that without running `apt update`, when I ran `apt install r-cran-spelling`, it installed hundreds of packages. After running `apt update`, it properly only installs the dependencies of r-cran-spelling.

**Comment by eddelbuettel** [comment]
Eyeballing the focal and jammy scripts next to each other, the differences are due to different reasons:
- the way the signing certifcate comes from the key server to `apt` changed
- with that the steps changed a little
- there was some issue with focal as I recall between the different repos involved (R binaries at CRAN, c2d4u as a base, r2u) which is where the `apt update` came from as I (vaguely) recall

So in short there are some non-obvious (but valid "at least at the time") differences here

**Comment by jdblischak** [comment]
I put together an example to demonstrate the behavior I am observing. I invited you to be a collaborator so that you can view the details in the GitHub Actions logs.

https://github.com/jdblischak/r2u-ci-example

There are 2 workflows. `no-update` follows the current steps detailed in `add_cranapt_jammy.sh`, and it installs many unnecessary packages. `with-update` adds the single update like this PR, and the correct packages are installed.

**Comment by eddelbuettel** [comment]
Looking at the two side by side -- this helps.   And side-by-side makes clear that there _is_ a bug in not having it. The preceding line(s) 19-20 add the actual r2u / cranapt repo, but without `apt update` we never acquire the information.  

What I do not understand is why/how is installs more packages.  That is just _weird_.  (Maybe pinning without a repo does, don't know.)

**Review by eddelbuettel — APPROVED** [review]
The info in https://github.com/jdblischak/r2u-ci-example is convincing, thank you!

**Comment by jdblischak** [comment]
> What I do not understand is why/how is installs more packages. That is just weird. (Maybe pinning without a repo does, don't know.)

I was also confused by this behavior. I thought that maybe something in the GitHub Ubuntu runner might be interfering, but looking at their r-base installation script, they delete everything afterwards.

https://github.com/actions/runner-images/blob/main/images/linux/scripts/installers/r.sh

Thanks for the merge! Heads up that I'm going to delete r2u-ci-example tomorrow, so if you'd like to experiment with it, please fork it to your account.

**Comment by eddelbuettel** [comment]
Thanks for the heads-up but I won't need it. I (currently) use a (test version of an) r2u-enabled r-ci script via the r-ci repo.
